api/v1alpha1/loadbalancer_types.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// LoadBalancer defines the load balancer policy to be applied.
+// +union
+type LoadBalancer struct {
+	// Type decides the type of Load Balancer policy.
+	// Valid RateLimitType values are
+	// "ConsistentHash",
+	// "LeastRequest",
+	// "Random",
+	// "RoundRobin",
+	//
+	// +unionDiscriminator
+	Type LoadBalancerType `json:"type"`
+	// ConsistentHash defines the configuration when the load balancer type is
+	// set to ConsistentHash
+	//
+	// +optional
+	ConsistentHash *ConsistentHash `json:"consistentHash,omitempty"`
```

Comment:
What happens if the type is `ConsistentHash` and the `ConsistentHash` is nil?

Reply:
this is what @zirain is referring to in https://github.com/envoyproxy/gateway/pull/2063#discussion_r1372418714
we can add CEL Validation to proactively check this

Reply:
can you open a ticket for this?

Reply:
raised https://github.com/envoyproxy/gateway/issues/2066